packages/@lwc/engine/src/framework/base-lightning-element.ts

```diff
@@ -341,19 +341,24 @@ BaseLightningElementConstructor.prototype = {
         const wrappedListener = getWrappedComponentsListener(vm, listener);
         vm.elm.removeEventListener(type, wrappedListener, options);
     },
-    setAttributeNS(ns: string | null, attrName: string, _value: string) {
+    hasAttribute(): string | null {
+        const elm = getLinkedElement(this);
+        // Typescript does not like it when you treat the `arguments` object as an array
+        // @ts-ignore type-mismatch
+        return elm.hasAttribute.apply(elm, arguments);
+    },
+    hasAttributeNS(): string | null {
+        const elm = getLinkedElement(this);
```

Comment:
after looking at the existing code, the lock/unlock is only in place to avoid the infinite loop when reflecting attributes and props. based on that assumption, it is safe to say that when observing attributes (has/get/etc), we don't need this. It is only needed when mutating the attributes.